allennlp/data/dataset_readers/stanford_sentiment_tree_bank.py

```diff
@@ -0,0 +1,125 @@
+    granularity : ``str``, optional (default = ``"5-class"``)
+        One of ``"5-class"``, ``"3-class"``, or ``"2-class"``, indicating the number
+        of sentiment labels to use.
+    lazy : ``bool``, optional, (default = ``False``)
+        Whether or not instances can be consumed lazily.
+    """
+    def __init__(self,
+                 token_indexers: Dict[str, TokenIndexer] = None,
+                 use_subtrees: bool = False,
+                 granularity: str = "5-class",
+                 lazy: bool = False) -> None:
+        super().__init__(lazy=lazy)
+        self._token_indexers = token_indexers or {'tokens': SingleIdTokenIndexer()}
+        self._use_subtrees = use_subtrees
+        allowed_granularities = ["5-class", "3-class", "2-class"]
+        if granularity not in allowed_granularities:
+            raise ConfigurationError("granularity is {}, but expected one of: {}".format(
+                    granularity, allowed_granularities))
+        self._granularity = granularity
+
+    @overrides
+    def _read(self, file_path):
+        with open(cached_path(file_path), "r") as data_file:
+            logger.info("Reading instances from lines in file at: %s", file_path)
+            for line in tqdm.tqdm(data_file.readlines()):
+                line = line.strip("\n")
+                if not line:
+                    continue
+                parsed_line = Tree.fromstring(line)
+                if self._use_subtrees:
+                    for subtree in parsed_line.subtrees():
+                        instance = self.text_to_instance(subtree.leaves(), subtree.label())
+                        if instance is None:
+                            continue
+                        yield instance
+                else:
+                    instance = self.text_to_instance(parsed_line.leaves(), parsed_line.label())
+                    if instance is None:
+                        continue
+                    yield instance
+
+    @overrides
+    def text_to_instance(self, tokens: List[str], sentiment: str = None) -> Instance:  # type: ignore
+        """
+        We take `pre-tokenized` input here, because we don't have a tokenizer in this class.
+
+        Parameters
+        ----------
+        tokens : ``List[str]``, required.
+            The tokens in a given sentence.
+        sentiment ``str``, optional, (default = None).
+            The sentiment for this sentence.
+
+        Returns
+        -------
+        An ``Instance`` containing the following fields:
+            tokens : ``TextField``
+                The tokens in the sentence or phrase.
+            label : ``LabelField``
+                The sentiment label of the sentence or phrase.
+        """
+        # pylint: disable=arguments-differ
+        text_field = TextField([Token(x) for x in tokens], token_indexers=self._token_indexers)
+        fields: Dict[str, Field] = {"tokens": text_field}
+        if sentiment is not None:
```

Comment:
it would be nice if there was a brief comment about the various granularities and the logic behind the conversion here